internal/cache/controller_obj_cache.go

```diff
@@ -1624,6 +1624,62 @@ func (c *AviObjCache) AviPopulateAllHttpPolicySets(client *clients.AviClient, cl
 	}
 	return httpPolicyData, result.Count, nil
 }
+func (c *AviObjCache) AviPopulateHttpPolicySetbyUUID(client *clients.AviClient, uuid string) error {
+
+	uri := "/api/httppolicyset/" + uuid
+	rawData, err := lib.AviGetRaw(client, uri)
+	if err != nil {
+		utils.AviLog.Warnf("Get uri %v returned err for httppolicyset %v", uri, err)
+		return err
+	}
+	httppol := models.HTTPPolicySet{}
+	err = json.Unmarshal(rawData, &httppol)
+	if err != nil {
+		utils.AviLog.Warnf("Failed to unmarshal httppolicyset data, err: %v", err)
+		return err
+	}
+
+	if httppol.Name == nil || httppol.UUID == nil {
+		utils.AviLog.Warnf("Incomplete http policy data unmarshalled, %s", utils.Stringify(httppol))
+		return errors.New("incomplete http policy data unmarshalled")
+	}
+
+	// Fetch the pgs associated with the http policyset object
+	var poolGroups []string
+	var pools []string
+	if httppol.HTTPRequestPolicy != nil {
+		for _, rule := range httppol.HTTPRequestPolicy.Rules {
+			if rule.SwitchingAction != nil {
+				val := reflect.ValueOf(rule.SwitchingAction)
+				if !val.Elem().FieldByName("PoolGroupRef").IsNil() {
+					pgUuid := ExtractUuid(*rule.SwitchingAction.PoolGroupRef, "poolgroup-.*.#")
+					pgName, found := c.PgCache.AviCacheGetNameByUuid(pgUuid)
+					if found {
+						poolGroups = append(poolGroups, pgName.(string))
+					}
+				} else if !val.Elem().FieldByName("PoolRef").IsNil() {
```

Comment:
Why are we populating either pools or poolgroup, not both ?

Reply:
@pkoshtavmware:In Avi controller, switching action can point to  either pool or PG.

Reply:
It is because SwitchingAction can be either to select HTTP_SWITCHING_SELECT_POOLGROUP or HTTP_SWITCHING_SELECT_POOL not both